core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/DefaultQuickPulsePingSender.java

```diff
@@ -49,39 +50,55 @@
     private final ApacheSender apacheSender;
     private final QuickPulseNetworkHelper networkHelper = new QuickPulseNetworkHelper();
     private String pingPrefix;
+    private String roleName;
+    private String instanceName;
+    private String machineName;
+    private String quickPulseId;
     private long lastValidTransmission = 0;
     private static volatile AtomicBoolean friendlyExceptionThrown = new AtomicBoolean();
 
-    public DefaultQuickPulsePingSender(ApacheSender sender, TelemetryConfiguration configuration, String instanceName, String quickPulseId) {
+    public DefaultQuickPulsePingSender(ApacheSender sender, TelemetryConfiguration configuration, String machineName, String instanceName, String roleName, String quickPulseId) {
         this.configuration = configuration;
         this.apacheSender = sender;
+        this.roleName = roleName;
+        this.instanceName = instanceName;
+        this.machineName = machineName;
+        this.quickPulseId = quickPulseId;
+
+        if (!LocalStringsUtils.isNullOrEmpty(roleName)) {
+            roleName = "\"" + roleName + "\"";
+        }
 
         pingPrefix = "{" +
                 "\"Documents\": null," +
                 "\"Instance\":\"" + instanceName + "\"," +
                 "\"InstrumentationKey\": null," +
                 "\"InvariantVersion\": 2," +
-                "\"MachineName\":\"" + instanceName + "\"," +
+                "\"MachineName\":\"" + machineName + "\"," +
+                "\"RoleName\":" + roleName + "," +
```

Comment:
same comment as above about removing conditional and changing this, primarily to match the other attributes

```suggestion
                "\"RoleName\":\"" + roleName + "\"," +
```

Reply:
This is a great question 😃 The problem here is that, without this conditional, if the value of `roleName` is `null` then it will become `roleName: "null"` in the resulting json, as in a string called "null" which is incorrect, we want it to be `roleName: null` so the service interprets this correctly.